src/Service/RequestService.php

```diff
@@ -784,6 +798,130 @@ public function proxyHandler(array $data, array $configuration, ?Source $proxy =
 
     }//end proxyHandler()
 
+    public function useRelayRating(array &$config): bool
+    {
+        $returnValue = true;
+        if (isset($config['query']['_federalization_relay_rating']) === true) {
+            $returnValue = $config['query']['_federalization_relay_rating'];
+
+            unset($config['query']['_federalization_relay_rating']);
+        }
+
+        return $returnValue;
+
+    }//end useRelayRating()
+
+    public function getFederationSources(array &$config, Collection $proxies): Collection
+    {
+        if (isset($config['query']['_federalization_use_sources']) === true && isset($config['query']['_federalization_exclude_sources']) === true) {
+            throw new Exception('Use of sources and exclusion of sources cannot be done in the same request');
+        }
+
+        $usedSourceIds     = [];
+        $excludedSourceIds = [];
+
+        if ((isset($config['query']['_federalization_use_sources']) === true && $config['query']['_federalization_use_sources'] === '*')
+            || (isset($config['query']['_federalization_exclude_sources']) === true && $config['query']['_federalization_exclude_sources'] === null)
+            || (isset($config['query']['_federalization_use_sources']) === false && isset($config['query']['_federalization_exclude_sources']) === false)
```

Comment:
This is not a very readable / quick to understand if statement, could you add a comment above it to explain what it does?
_and maybe check again if you made any typos/mistakes or not, cause I am not sure what the 2nd line is for_